### PR TITLE
Added object resolver

### PR DESCRIPTION
**Pull request by veblush**
- Added object resolver to iterate multiple objects for benchmark
- Added detailed info such as bucket, object, and size to the verbose reported
- Update gRPC & GCSIO to the latest

### CODE REVIEW

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java [hunk 1]
```diff
@@ -202,11 +174,36 @@ private void makeRandomMediaRequest(ResultTable results) throws IOException {
...
+
+    for (int i = 0; i < args.calls; i++) {
+      long start = System.currentTimeMillis();
+      String object = objectResolver.Resolve(threadId, i);
+      URI uri = URI.create("gs://" + args.bkt + "/" + object);
+      WritableByteChannel writeChannel = gcsfs.create(uri);
+      ByteBuffer buff = ByteBuffer.wrap(randBytes);
+      writeChannel.write(buff);
+      writeChannel.close();
+      // write operation is async, need to call close() to wait for finish.
```
Comment (mohanli-ml) [inline end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java:202]: Should this comment be moved before L207?

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GrpcClient.java [hunk 2]
```diff
@@ -148,13 +150,13 @@ public void startCalls(ResultTable results) throws InterruptedException {
       try {
         switch (args.method) {
           case METHOD_READ:
-            makeReadObjectRequest(channel, results);
+            makeReadObjectRequest(channel, results, 1);
```
Comment (mohanli-ml) [inline end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GrpcClient.java:153]: Add a comment /*threatId*/?

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java [hunk 3]
```diff
@@ -84,28 +80,40 @@ public void startCalls(ResultTable results) throws InterruptedException, IOExcep
...
             for (int i = 0; i < args.threads; i++) {
-              Runnable task =
-                  () -> {
-                    try {
-                      makeMediaRequest(results);
-                    } catch (IOException e) {
-                      e.printStackTrace();
-                    }
-                  };
+              int finalI = i;
```
Comment (mohanli-ml) [inline end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java:83]: Could you explain what is the meaning of this name `finalI`? It seems this local variable is not necessary.

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java [hunk 4]
```diff
@@ -142,53 +147,20 @@ private void makeMediaRequest(ResultTable results) throws IOException {
...
-            GoogleCloudStorageFileSystemOptions.builder().setCloudStorageOptions(gcsOpts).build());
+  private void makeRandomMediaRequest(ResultTable results, int threadId) throws IOException {
+    GoogleCloudStorageFileSystem gcsfs = new GoogleCloudStorageFileSystem(creds,
+        GoogleCloudStorageFileSystemOptions.builder().setCloudStorageOptions(gcsOpts).build());
 
     Random r = new Random();
 
-    URI uri = URI.create("gs://" + args.bkt + "/" + args.obj);
-
+    String object = objectResolver.Resolve(threadId, 0);
```
Comment (mohanli-ml) [inline end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java:162]: Maybe add a comment "/* objectId = */"?

**Review by mohanli-ml — APPROVED** [review]
Thanks! Some small comments if you think it applies.

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java (hunk 3 above)
Comment (veblush) [reply end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java:83]: This is to capture the value of `i`. https://stackoverflow.com/questions/54340101/enhanced-for-loop-and-lambda-expressions

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GrpcClient.java (hunk 2 above)
Comment (veblush) [reply end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GrpcClient.java]: Done.

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java (hunk 4 above)
Comment (veblush) [reply end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java]: Done.

File: end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java (hunk 1 above)
Comment (veblush) [reply end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java:216]: > Could you move this comment before 201?
Done.

> And should L197 to L202 be put inside the for loop or outside?
What do you mean by that?

**Comment by veblush** [comment]
Getting this merged because I need this but I'll revisit your comment if required!

**Pull request merged**